portfolio/src/main/webapp/script.js

```diff
@@ -101,16 +98,56 @@ function deleteMesage(message) {
 
 function loadCV() {
   const CVContainer = document.getElementById('cv-container');
-  CVContainer.innerText = "Opening...";
+  CVContainer.innerText = 'Opening...';
+}
+
+function loadNicknameForm() {
+  fetch('/user-info').then(response => response.json()).then((userInfo) => {
+    var isLoggedIn = userInfo["is-logged-in"];
+
+    if (!isLoggedIn) {
+      document.getElementById('nickname-container').hidden = true;
+      document.getElementById('log-container').innerHTML = `<p>Want to choose a nickname? Please, <a href=${userInfo["login-url"]}>login</a></p>`;
+    } else {
+      document.getElementById('nickname-container').hidden = false;
+      document.getElementById('log-container').innerHTML = `<p>You can <a href=${userInfo["logout-url"]}>logout</a></p>`;
+      
+      var nickname = userInfo["nickname"];
+      if (nickname != '' && nickname != null) {
+        document.getElementById('nickname').value = nickname;
+      }
+    }
+  });
+}
+
+function loadMessageForm() {
+  fetch('/user-info').then(response => response.json()).then((userInfo) => {
```

Comment:
You are fetching this multiple times in the code. Better fetch it only once and pass the data to the methods.